### ISSUE TITLE
[NaN] Support NaN values in loader

### ISSUE THREAD

**Issue by blaggacao** [odoo/odoo#21742]
`float('nan')`, as defined in IEEE 754 are a common way for data manipulation tools, such as `numpy` or `pandas` to treat non existent value.
Please include these into the loader function to make Odoo's API mor plug-and-play.

/cc @rim-odoo

**Comment by Yenthe666** [comment]
@qsm-odoo and @ged-odoo absolutely of interest for you two!

**Comment by rim-odoo** [comment]
Not sure why I'm pinged here... but please can you provide your use case?

**Comment by blaggacao** [comment]
@rim-odoo Your avatar transmits deep technical authority :smile: 

### Use case:
ERP MIGRATION / Data integration towards Odoo coming from an ETL solution (usually based on `pandas` or `numpy` arrays or somthing similar for data crunching).

### Observation:
`void` values are internally represented as `NaN`s. Connecting to the loader api is usually done by some kind of row-by-row or batch exporter who directly reads from the dataframes.

### Problem:
Interfering on this last mile is kind of error prone as you need (sometimes complicated or slow on local machine) interception logic on differently shaped arrays, usually tuples, just before the POST request.

It's probably a first world problem, but technically interesting. :smile:

**Comment by rim-odoo** [comment]
I would manage to export these `NaN`s as `False`/`0` or (other falsy/0 value) before importing them in Odoo, but I agree, it is one step more in the pipeline.

Actually you are trying to adapt Odoo to your data format instead of adapting your data format to Odoo :)

**Comment by blaggacao** [comment]
Your half way correct, as "my format" is the standard way of dealing with '' values in data science applications...

So I would rephrase: I'm actually trying to make Odoo's interface more friendly to a wider variety of applications. :)

**Comment by blaggacao** [comment]
Cleverest check fo `NaN`s, I've found:

```python
(x != x) == True
```

**Comment by blaggacao** [comment]
If a t some point someone might search this:

## My Solution
```python
def convert(item):
    try:
        return item.item()  # normalize numpy objects
    except:
        if item != item:  # Item is a nan
            return ''
        else:
            return item
```

**Comment by mart-e** [comment]
Hi there,

After all these years, this is still not a frequent request. Let's not complicate the loader with weird usecases.

**Comment by TadeuszKarpinski** [comment]
still an issue

**Comment by salvorapi** [comment]
still an issue
